structs/ctl/types.go

```diff
@@ -153,6 +157,32 @@ type SigningCertMsg struct {
 	WebsocketControl
 }
 
+type dataType string
+
+const (
+	SourcesRequest dataType = "sources"
+)
+
+type DataReadRequest struct {
+	Type        dataType        `json:"data_type"`
+	CommunityID string          `json:"community_id"`
+	Token       string          `json:"token"`
+	Body        json.RawMessage `json:"body"`
+}
+
+type DataReadResponse struct {
+	Error error  `json:"error"`
```

Comment:
Does this error field ever get populated? The `api.WriteError` helper sets the Error field in the ResponseWrapper struct, so we don't have to worry about it. Also what is Data in this struct?

Reply:
oh oops i'll remove error. Data is the actual data from the read request